### PR TITLE
fix copy button invalid & copy button (invisible) duplication bug in chatbot

### LINKED ISSUE #4324: Copy of code blocks does not work for LAN access
### Describe the bug

When I access my application on the LAN and click the copy button in the code block, an error is reported
If I access localhost, the problem will no longer exist
Please fix it as soon as possible
image

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

.launch(server_name="0.0.0.0" 

### Screenshot

_No response_

### Logs

```shell
properties of undefined (reading 'writeText')
    at HTMLButtonElement.<anonymous> (ChatBot.svelte:57:25)
(匿名) @ ChatBot.svelte:57
ChatBot.svelte:5
```

### System Info

```shell
no
```

### Severity

annoying

### PR DESCRIPTION
**Pull request by binary-husky**
# Bug

- The copy button is added multiple times when the update event is called in chatbot.
- The copy button is not working when accessing remotely (not local + not https), closes #4324 

# Influence

Slow down the browser, but cannot be seen visually.

# Fix 
Detect whether a copy button is already in the block.
image

# Before fix
image

# After fix
image

### CODE REVIEW

**Comment by abidlabs** [comment]
Thanks @binary-husky! Would you mind adding an entry to the CHANGELOG? 

@dawoodkhan82 can you review?

**Comment by binary-husky** [comment]
> Thanks @binary-husky! Would you mind adding an entry to the CHANGELOG?
> 
> @dawoodkhan82 can you review?

Sure
https://github.com/gradio-app/gradio/pull/4350/commits/82ecc5735ba15f1b95a38040432d853e6720b312

**Comment by pngwn** [comment]
I need to make some tweaks to this before it is merged. Will do that in the next few hours.

File: js/chatbot/src/Copy.svelte [hunk 1]
```diff
@@ -16,9 +16,30 @@
...
 			await navigator.clipboard.writeText(value);
 			copy_feedback();
+		} else {
+			// Use the 'out of viewport hidden text area' trick
+			const textArea = document.createElement("textarea");
+			textArea.value = value;
+
+			// Move textarea out of the viewport so it's not visible
+			textArea.style.position = "absolute";
+			textArea.style.left = "-999999px";
```
Comment (Keldos-Li) [inline js/chatbot/src/Copy.svelte:30]: What about
```js
textArea.style.display = "none";
```

---

I was just about to fix the issue of duplicated buttons. Lucky to see this pr.

Comment (binary-husky) [reply js/chatbot/src/Copy.svelte:30]: 😄

**Pull request merged**